packages/http/src/models/Challenge.ts

```diff
@@ -150,19 +152,23 @@ const constructPath = (path: HTTPPathObject[]): Path[] => {
     }
     return {
       witnesses: pathObject.witnesses.map(
-        (witness) => ({
-          timestamp: witness.timestamp,
-          snr: witness.snr,
-          signal: witness.signal,
-          packetHash: witness.packet_hash,
-          owner: witness.owner,
-          location: witness.location,
-          isValid: isValidWitness(witness),
-          gateway: witness.gateway,
-          frequency: witness.frequency,
-          datarate: witness.datarate,
-          channel: witness.channel,
-        } as Witness),
+        (witness) => {
+          let witnessObject = {
+            timestamp: witness.timestamp,
+            snr: witness.snr,
+            signal: witness.signal,
+            packetHash: witness.packet_hash,
+            owner: witness.owner,
+            location: witness.location,
+            isValid: isValidWitness(witness),
+            gateway: witness.gateway,
+            frequency: witness.frequency,
+            datarate: witness.datarate,
+            channel: witness.channel,
+          } as Witness
+          if (!isValidWitness(witness)) witnessObject = { ...witnessObject, invalidReason: witness.invalid_reason }
```

Comment:
I think you could do this without defining and mutating an intermediate object:

```js
{
  timestamp: witness.timestamp,
  ...(!isValidWitness(witness)) && { invalidReason: witness.invalid_reason },
}
```

Also, is the `isValidWitness` function call necessary? or could you just check to see if `invalid_reason` is present in the json?

Reply:
oh yeah that's way better, I tried to do something like that but couldn't remember the syntax for it, nice! and yeah will just check for witness.invalid_reason instead, good call 👍